cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -347,7 +347,8 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.entityOperatorUserOpRoleBindingForRole())
                 .compose(state -> state.entityOperatorTopicOpAncillaryCm())
                 .compose(state -> state.entityOperatorUserOpAncillaryCm())
-                .compose(state -> state.entityOperatorSecret(this::dateSupplier))
+                .compose(state -> state.entityTopicOperatorSecret(this::dateSupplier))
+                .compose(state -> state.entityUserOperatorSecret(this::dateSupplier))
```

Comment:
The upgrade tests seem to pass fine. But I do not see any place where we delete the old certificate which is not completely nice since it would basically stay there forever. It would be nice if we can delete it as well here. Basically just adding some step with `secretOperations.reconcile(..., ..., ..., null)` and some comment explaining why are we deleting it.

Reply:
@scholzj I looked at the code again, and it seems like we have the code to delete the certificates when the topic Operator or user Operator is deleted. Check this:
```
Future<ReconciliationState> entityTopicOperatorSecret(Supplier<Date> dateSupplier) {
            return updateCertificateSecretWithDiff(EntityTopicOperator.secretName(name), entityOperator == null || entityOperator.getTopicOperator() == null ? null : entityOperator.getTopicOperator().generateSecret(clusterCa, isMaintenanceTimeWindowsSatisfied(dateSupplier)))
        .map(changed -> {
            existingEntityTopicOperatorCertsChanged = changed;
            return this;
        });
}
```

Basically the Secret object will be null, whenever the topic Operator is removed. Within the secretOperations.getAsync() method, there is a call to the secretOperations.reconcile(). -> This method does take a null as the last argument.

I also manually run this test: `testRemoveUserOperatorFromEntityOperator`. I do see the secret for the user operator was deleted and then recreated in the test.

Does this make sense to you?

Reply:
I meant this

`entityOperator == null || entityOperator.getTopicOperator() == null ? null : entityOperator.getTopicOperator().generateSecret(clusterCa, isMaintenanceTimeWindowsSatisfied(dateSupplier))`

The secret will become null if the topic operator is not defined. Otherwise, try to generate the secret

Reply:
That is correct. But what I meant here is something else. If you run Strimzi 0.26, it will maintain a secret named `<my-cluster>-entity-operator-certs` with certificate inside. Now once your PR is merged and users upgrade to it, there will be two new secrets: `<my-cluster>-entity-user-operator-certs` and `<my-cluster>-entity-topic-operator-certs`. But nobody will delete the old `<my-cluster>-entity-operator-certs` secret which will not be used anywhere but will still exist. Or is the old secret somewhere deleted and I just missed it?

Reply:
I see what you mean now:-). You are right! We don't clean up the old certificate now, which we should. Let me add it in this PR.